lib/services/db.js

```diff
@@ -148,146 +107,15 @@ function list(options) {
- * @param {object} op
- * @param {string} op.type
- * @param {string} op.key
- * @param {*}      op.value
- * @returns {object}
- */
-function addToHookOps(hookOps, op) {
-  hookOps[op.type].push(op);
-  return hookOps;
-}
-
-/**
- *
- * @param {string} type
- * @param {string} key
- * @param {*} value
- * @returns {[{}]}
- */
-function singleOpToBatch(type, key, value) {
-  return [{type: type, key: key, value: value}];
-}
-
-/**
- * reduce database operations to key-value arrays sorted by operation type
- * defines what the plugin hooks expect to receive
- * @param {string} method
- * @param {Array} args    args from put, del, or batch
- * @returns {{put: [{key: value}], del: [{key: value}]}}
- */
-function dbOpsToHookOps(method, args) {
-  var hookOps = {
-      put: [],
-      del: []
-    },
-    batchOps = method === 'batch' ? args[0] : singleOpToBatch(method, args[0], args[1]);
-
-  return _.reduce(batchOps, addToHookOps, hookOps);
-}
-
-/**
- *
- * @param {string} method
- * @param {Array} args
- */
-function triggerPlugins(method, args) {
-  const hookOps = dbOpsToHookOps(method, args),
-    // assumes page publish batches always have page put as the last op
-    lastPutKey = _.get(_.last(hookOps.put), 'key'),
-    isPublishedPageBatch = method === 'batch' && hookOps.put.length > 1 &&  _.includes(lastPutKey, '/_pages/') && _.endsWith(lastPutKey, publishedVersionSuffix);
-
-  if (hookOps.put.length) {
-    if (isPublishedPageBatch) {
-      plugins.executeHook('publish', {
-        uri: lastPutKey,
-        ops: hookOps.put
-      });
-    } else {
-      plugins.executeHook('save', hookOps.put);
-      bus.publish('save', JSON.stringify(hookOps.put));
-    }
+function registerStorage(storage) {
+  for (let action in storage) {
+    module.exports[action] = storage[action];
   }
-  if (hookOps.del.length) {
-    plugins.executeHook('delete', hookOps.del);
-    bus.publish('delete', JSON.stringify(hookOps.del));
-  }
-}
-
-/**
- * notifies outside listeners through eventify and amphora plugins if operation is successful
- * @param {string} method
- */
-function exposeMethodToOutsideListeners(method) {
-  module.exports[method] = _.wrap(module.exports[method], function (fn) {
-    const args = _.slice(arguments, 1);
-
-    return fn.apply(module.exports, args).then(function (result) {
-      module.exports.trigger.apply(module.exports, [method].concat(args)); // Eventify
-      triggerPlugins(method, args);
-      return result;
-    });
-  });
 }
 
-module.exports.get = get;
-module.exports.put = put;
-module.exports.del = del;
 module.exports.list = list;
-module.exports.batch = batch;
-module.exports.clear = clear;
-module.exports.getDB = function () { return db; };
-module.exports.setDB = function (value) { db = value; };
 module.exports.pipeToPromise = pipeToPromise;
+module.exports.registerStorage = registerStorage;
 
-Eventify.enable(module.exports);
-
-_.each(['put', 'batch', 'del'], exposeMethodToOutsideListeners);
```

Comment:
what outside listeners did we have (if any)?

Reply:
None, we just never deprecated it 😬